### ISSUE TITLE
`shortcuts` should not be a publicly accessible attribute

### ISSUE THREAD

**Issue by kotfu** [python-cmd2/cmd2#683]
We have some inconsistencies in our api and our documentation. In 0.9.12 we added ``shortcuts`` as an optional argument to `__init__()`, and changed the initialization code associated with these shortcuts. This created the following issues:

- the changelog said nothing about disabling the `.shortcuts` attribute. Prior to 0.9.12 you could modify that attribute to change the shortcuts. After 0.9.12, the only way to change the shortcuts is to pass them to `__init__()`
- What's the point of the `shortcuts` attribute on a `cmd` object if the caller has to pass it in, and it's read only? When `cmd2` code needs it, can get it from the `statement_parser`, and code that instantiates `cmd2.Cmd()` or a subclass thereof already know it because they passed it in.
- the docstring for the __init__ method says you should pass shortcuts as a dictionary, and then the initialization code processes those and stores a tuple in the `shortcuts` attribute (which has a property to access it). It seems inconsistent to pass a dictionary and access a tuple.
- The DEFAULT_SHORTCUTS variable should really be in `constants.py`, not `cmd2.py`

**Comment by tleonhardt** [comment]
Well shoot, I did a much better job discussing these changes in the PR #648 than I did in the changelog!  

One decision that was made was to intentionally make shortcuts, multiline commands, and terminators immutable where they get passed in to __init__() at startup, then get passed the StatementParser and within that they get converted to an immutable tuple.  

The point was to break out certain things that maybe should be constant over time for a given application and not be user-modifiable during runtime and enforce their immutability.

The concept behind passing a dictionary was to make it easy for devs to progressively build up shortcuts in a mutable type if they so chose before instantiating a cmd2.Cmd instance.  But maybe your right and it is unnecessarily confusing.  

Completely agree that the default shortcuts should be in constants.py.

I’m open to whatever changes you think make the most sense to help achieve a more consistent API.
